pinot-plugins/pinot-minion-tasks/pinot-minion-builtin-tasks/src/main/java/org/apache/pinot/plugin/minion/tasks/BaseMultipleSegmentsConversionExecutor.java

```diff
@@ -352,6 +336,59 @@ public List<SegmentConversionResult> executeTask(PinotTaskConfig pinotTaskConfig
     }
   }
 
+  private int parallelDownloadAndUntarSegments(String[] downloadURLs, String[] segmentNames, String tableNameWithType,
+      String taskType, File tempDataDir, List<File> inputSegmentDirs)
+      throws Exception {
+    AtomicInteger numRecords = new AtomicInteger(0);
+    List<Future<Void>> futures = new ArrayList<>();
+
+    int nThreads = Math.min(downloadURLs.length, Runtime.getRuntime().availableProcessors());
+    ExecutorService executorService = Executors.newFixedThreadPool(nThreads);
```

Comment:
after adding the configs, can we add / modify a test case that parallelly downloads multiple segments ? 

because its possible many sub methods like downloadSegmentToLocalAndUntar / reportSegmentDownloadMetrics aren't thread safe and a test case can help catch such issues 

Even if they're thread safe now, a test case that downloads multiple segemnts parallely can help catch other issues that might exist now or in future